A better implementation of the Ollama component

yamonkjd opened the pull request:
A draft to synchronize the model using the latest Langflow architecture and to improve it according to the latest Langchain specifications.

ogabrielluiz commented:
Hey  @yamonkjd 
How are you? 

Thanks for all the help.

We'd love to get Ollama to an awesome level. Do you plan on having this ready soon? I could take a crack at it if you are swamped today.

yamonkjd commented:
> swamped

Hello,

The testing for Ollama's Keepalive was not normal, so we are still working on it.

We will complete it and reach out to you within today.

yamonkjd commented:
> Hey @yamonkjd How are you?
> 
> Thanks for all the help.
> 
> We'd love to get Ollama to an awesome level. Do you plan on having this ready soon? I could take a crack at it if you are swamped today.

Hello,

The testing for Ollama's Keepalive was not normal, so we are still working on it.

We will complete it and reach out to you within today.

yamonkjd commented:
I removed the callback element as it seems it will be replaced with Langflow's management methods. I also removed caching as its operation was not guaranteed (I think this will likely be integrated into Langflow as well).
Before the request, I performed Make lint and confirmed that there are no conflicts.

However, since it is not the initially written version, it seems there is an error from Langflow itself (Openapi key).
It seems there are no problems in the main code, and there have been many requests, so I request a merge.

yamonkjd commented:
This proposal was simply made because we need it. If there is a better implementation, it can be discarded or used simply for reference. We have also been considering implementing interaction features using Ollama's API, which would allow us to add models directly or specify model files ourselves (such as selecting a model from Ollama's official model list and installing it on an instance).

ogabrielluiz reviewed (approved):
LGTM 

Should we merge it?
